File: invesdwin-scripting-parent/invesdwin-scripting-rust/invesdwin-scripting-rust-runtime-contract/src/test/java/de/invesdwin/scripting/rust/runtime/contract/callback/ParametersAndReturnsTestFloat.rs
println!("getFloat");
let getFloat: f64 = callback("getFloat", &[]);
println!("getFloat: {}", getFloat);
callback_void("setFloat", &[param(getFloat)]);

println!("getFloatVector");
let getFloatVector: Vec<f64> = callback("getFloatVector", &[]);
println!("getFloatVector: {:?}", getFloatVector);
callback_void("setFloatVector", &[param(getFloatVector)]);

println!("getFloatVectorAsList");
let getFloatVectorAsList: Vec<f64> = callback("getFloatVectorAsList", &[]);
println!("getFloatVectorAsList: {:?}", getFloatVectorAsList);
callback_void("setFloatVectorAsList", &[param(getFloatVectorAsList)]);

println!("getFloatMatrix");
let getFloatMatrix: Vec<Vec<f64>> = callback("getFloatMatrix", &[]);
println!("getFloatMatrix: {:?}", getFloatMatrix);
callback_void("setFloatMatrix", &[param(getFloatMatrix)]);

println!("getFloatMatrixAsList");
let getFloatMatrixAsList: Vec<Vec<f64>> = callback("getFloatMatrixAsList", &[]);
println!("getFloatMatrixAsList: {:?}", getFloatMatrixAsList);
callback_void("setFloatMatrixAsList", &[param(getFloatMatrixAsList)]);

File: invesdwin-scripting-parent/invesdwin-scripting-haskell/invesdwin-scripting-haskell-runtime-contract/src/test/java/de/invesdwin/scripting/haskell/runtime/contract/callback/ParametersAndReturnsTestFloat.hs
putStrLn ( show ( "getFloat" ) )
getFloat :: Float <- callback "getFloat" []
putStrLn ( show ( getFloat ) )
callback "setFloat" [param getFloat] :: IO ()

putStrLn ( show ( "getFloatVector" ) )
getFloatVector :: [Float] <- callback "getFloatVector" []
putStrLn ( show ( getFloatVector ) )
callback "setFloatVector" [param getFloatVector] :: IO ()

putStrLn ( show ( "getFloatVectorAsList" ) )
getFloatVectorAsList :: [Float] <- callback "getFloatVectorAsList" []
putStrLn ( show ( getFloatVectorAsList ) )
callback "setFloatVectorAsList" [param getFloatVectorAsList] :: IO ()

putStrLn ( show ( "getFloatMatrix" ) )
getFloatMatrix :: [[Float]] <- callback "getFloatMatrix" []
putStrLn ( show ( getFloatMatrix ) )
callback "setFloatMatrix" [param getFloatMatrix] :: IO ()

putStrLn ( show ( "getFloatMatrixAsList" ) )
getFloatMatrixAsList :: [[Float]] <- callback "getFloatMatrixAsList" []
putStrLn ( show ( getFloatMatrixAsList ) )
callback "setFloatMatrixAsList" [param getFloatMatrixAsList] :: IO ()
